test/erlcloud_ecs_tests.erl

```diff
@@ -174,14 +174,14 @@ create_service_input_tests(_) ->
         },
         \"deployments\": [
           {
-            \"createdAt\": 1430326887.362,
+            \"createdAt\": 1430326887362,
```

Comment:
It looks like these should be left as-is. Amazon is real weird with these, and they do it with a number of APIs. probably the spec should be `number()` rather than `float()`, etc.

https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_CreateService.html#API_CreateService_Example_2_Response

Reply:
Sure, it seems difficult to rely on `erlcloud`'s reply though, since UNIX timestamp 1430326887.362 means `Sat Jan 17 1970 14:18:46`, while 1430326887362 means `Wed Apr 29 2015 18:01:27`. Is there a function to convert this to a readable format, somewhere in `erlcloud`'s API?

Reply:
In any case, I'm reverting the change.

Reply:
541f36d

Reply:
> Sure, it seems difficult to rely on `erlcloud`'s reply though, since UNIX timestamp 1430326887.362 means `Sat Jan 17 1970 14:18:46`, while 1430326887362 means `Wed Apr 29 2015 18:01:27`. Is there a function to convert this to a readable format, somewhere in `erlcloud`'s API?

I am a little confused here—or rather, I think you might be confusing epoch seconds and epoch milliseconds.

I got this from [epochconverter.com](https://www.epochconverter.com/batch):

Epoch date | Human-readable date (GMT)
-- | --
1430326887 | Wed, 29 Apr 2015 17:01:27 +0000
1430326887362 | 03 April 47295 03:16:02 GMT-5

(note: had to use a different converter for `1430326887362` since it refuses to convert it)

I don't think there should be any need to have a utility in erlcloud—it looks like you get back an epoch timestamp (in seconds) as a float, with 2-3 decimal places of precision if you want to account for that. It's OK as an exercise left for the reader, IMO,

Reply:
You're more or less right about "you might be confusing epoch seconds and epoch milliseconds"; I know the _epoch_ concept, but I messed up converting the value from the test files.

I did use `new Date(_)` from the JavaScript console, but forgot the base unit for that is milliseconds, so should have multiplied by 1000 [first error]. Since my original change to the (test) code was simply removing the `.` I further got convinced that AWS was wrong, which it kind of is, depending on the "point of view" [second error].

The correct way is
```
> new Date(1430326887000)
< Wed Apr 29 2015 18:01:27 GMT+0100
```

In any case, UNIX timestamp (unless AWS redefines it - I can't find references to how they define the **Timestamp** type) is a concept in seconds[1]. I've never heard about _epoch milliseconds_, but am willing to accept it in the scope of this Pull Request.

[1] for lack of a better alternative, here is the definition I'm used to: https://en.wikipedia.org/wiki/Unix_time ("It is the number of seconds...").

To be even more specific, the reference I use is, in Linux,
```
date -j -f "%a %b %d %T %Z %Y" "`date`" "+%s"
```

> It's OK as an exercise left for the reader, IMO.



Would it be acceptable to introduce a comment stating what is obvious for you but might not be to the consumer (e.g. `% epoch timestamp, in seconds`), or do you prefer to leave it as is?

Reply:
Sure, a comment is fine, but I think the intent is for the documentation to [link to the AWS API Reference](https://github.com/erlcloud/erlcloud/blob/541f36d0f5b6c9ac4c81ecccfe483bd9a469923a/src/erlcloud_ecs.erl#L1264) corresponding to each function, so that you can see what values AWS gives you.